### DISCUSSION TITLE
Atom highlight

### DISCUSSION THREAD

**Discussion by kgulyayeva** [jsme-editor/jsme-editor.github.io#6] [Q&A] [answered]
I am impressed by JSME! I have read that it is possible to highlight (or mark) the atoms or the bonds you choose. I wonder whether it is possible to make the following: I want to embed JSME on my Internet page, and then the user draws a molecular structure, then JSME generates a MOL file, then some substructure search is done based on the MOL file data (I want to search for functional groups),
and then (when the search is done) the user can see several **functional groups highlighted in different colors** in the JSME editor (several atoms that comprise the functional group should have one background color, other functional groups have different background color, and so on, if functional groups are present). I want to make this for educational purposes. I have some Javascript skills, and I have managed to embed JSME to html page, but I am struggling on how to make atoms comprising a functional group highlighted.
If you can give me a piece of advice on that, please, drop me a line . Have a great day!

Sincerely,
Karina
kgulyayeva@gmail.com

**Comment by kgulyayeva** [comment]
I got the task of highlighting atoms partially done. I also wonder if there is a way to mark atoms in more than 6 colors? Since there could be more than 6 functional groups in the molecule. Thanks in advance!

Sincerely,
Karina.

**Comment by jsme-editor** [comment, marked as answer]
For more than 6 background colors, have a look at this example:

https://jsme-editor.github.io/dist/JSME_functional_group_coloring.html

**Reply by kgulyayeva** [reply]
Thank you very much for this post! It helped me a lot!

Sincerely,
Karina
